src/core/lib/iomgr/error.h

```diff
@@ -127,7 +127,11 @@ typedef enum {
 #define GRPC_ERROR_OOM ((grpc_error*)2)
 #define GRPC_ERROR_RESERVED_2 ((grpc_error*)3)
 #define GRPC_ERROR_CANCELLED ((grpc_error*)4)
-#define GRPC_ERROR_SPECIAL_MAX GRPC_ERROR_CANCELLED
+#define GRPC_ERROR_RESERVED_3 ((grpc_error*)5)
+// GRPC_ERROR_EOS is used only by transports for send ops that failed because
+// the stream was closed for writes.
+#define GRPC_ERROR_EOS ((grpc_error*)6)
```

Comment:
This is probably okay for now, but I think it's worth considering how we want this to work in the future.  I'd like to eventually replace `grpc_error` with `absl::Status`.  We can add a custom payload to the `absl::Status` to include additional gRPC-specific debugging information, and we can have some utility functions that make it easy to construct a tree of `Status` objects the same way we do with errors today, but I don't think we'll have the ability to create special values for these kinds of cases anymore.  How would you handle this case (or any of the existing cases, like CANCELLED) in such a world?

Reply:
I guess one way would be to verify that that error does not have any children

Reply:
That doesn't sound quite right.  Consider the case where you're using `grpc_error_add_child()` to wrap an error on the way back up the stack.  Currently, that function treats `GRPC_ERROR_NONE` specially, but in this case you would want to also treat `GRPC_ERROR_EOS` specially, since neither of those values actually represent an error.

Actually, thinking about this further, because this condition is not actually an error, it does seem wrong to indicate it via a `grpc_error` value.  Doing it this way will basically mean that any filter that intercepts `on_complete` on the way back up the stack will need to know to ignore both `GRPC_ERROR_NONE` and `GRPC_ERROR_EOS`.

What was wrong with representing this via the `stream_write_closed` field in the transport API?  That actually seems like a much more natural representation.

Reply:
Well, GRPC_ERROR_EOS is an error since the ops need to be failed because of this error. It's just that we do not want to cancel the call so that we can avoid overwriting the status. 

There was nothing wrong with the stream_write_closed representation. I thought that GRPC_ERROR_EOS would be a better way since we do not need to add a field to the batch and we can simply use a special type of error to achieve the same purpose (similar to an int return value with varying error codes.)

Reply:
reverted the change.